Jules.md

```diff
@@ -0,0 +1,108 @@
+# Jules Onboarding Guide for `firebase-android-sdk`
+
+This guide provides essential information for working within the `firebase-android-sdk` repository.
+
+## Table of Contents
+- [Environment Setup](#environment-setup)
+- [Testing](#testing)
+- [API Surface](#api-surface)
+- [Best Practices](#best-practices)
+- [Common Patterns](#common-patterns)
+- [External Dependencies](#external-dependencies)
+- [Updating this Guide](#updating-this-guide)
+
+---
+
+## Environment Setup
+
+To work with this repository, the Android SDK must be installed. Use the `sdkmanager` command-line tool for this purpose.
+
+1.  **Install Android SDK Command-Line Tools**:
+    - If not already installed, download the command-line tools from the [Android Studio page](https://developer.android.com/studio#command-line-tools-only).
+    - Create a directory for the Android SDK, e.g., `android_sdk`.
+    - Unzip the downloaded package. This will create a `cmdline-tools` directory. Move this directory to `android_sdk/cmdline-tools/latest`.
+    - The final structure should be `android_sdk/cmdline-tools/latest/`.
+
+2.  **Install required SDK packages**:
+    - Use `sdkmanager` to install the necessary platforms, build tools, and other packages. For example:
+      ```bash
+      # List all available packages
+      sdkmanager --list
+
+      # Install platform tools and the SDK for API level 33
+      sdkmanager "platform-tools" "platforms;android-33"
+
+      # Accept all licenses
+      sdkmanager --licenses
+      ```
+    - Refer to the specific requirements of the project to determine which packages to install.
+
+3.  **Configure for integration tests**:
+    - To run integration tests, a `google-services.json` file is required.
+    - Place this file in the root of the repository.
+
+4.  **Install NDK for specific projects**:
+    - Some projects, like `firebase-crashlytics-ndk`, require a specific version of the Android NDK. You can install it using `sdkmanager`. For example, to install NDK version 21.4.7075529, you would run `sdkmanager "ndk;21.4.7075529"`. Always refer to the project's `README.md` for the exact version required.
+
+---
+
+## Testing
+
+This repository uses two main types of tests:
+
+1.  **Unit Tests**:
+    - These tests run on the local JVM.
+    - To execute unit tests for a specific project, run:
+      ```bash
+      ./gradlew :<firebase-project>:check
```

Comment:
Is there a difference between check and test?

Reply:
Check is more comprehensive than test https://stackoverflow.com/questions/50104666/difference-between-test-and-check-tasks-in-gradle